pkg/chunkenc/memchunk.go

```diff
@@ -152,6 +152,94 @@ func (hb *headBlock) serialise(pool WriterPool) ([]byte, error) {
 	return outBuf.Bytes(), nil
 }
 
+// CheckpointBytes is used for WAL checkpointing
+func (hb *headBlock) CheckpointBytes(version byte) ([]byte, error) {
```

Comment:
I think this deserve a better comment as to what it does and not who uses it.

Not sure yet how it uses but have you thought about compression? what's you take ?

Reply:
> I think this deserve a better comment as to what it does and not who uses it.

👍 

> Not sure yet how it uses but have you thought about compression? what's you take ? What's the difference with serialise(pool WriterPool) ([]byte, error) ?

I have thought about it, but I omitted compression for this PR for a few reasons:
1) simplicity - we can easily add it in a followup
2) I'm still experimenting and I haven't found this to be a bottleneck yet. There's somewhat of a balancing act between cpu contention compressing records/checkpoints and disk contention writing them to. That being said, I think adding compression here will be very beneficial.

> I would prefer that you implement io.WriterTo here. This has the advantage of giving the power to re-use the buffer to the caller.

👍

Reply:
Sounds good to me.

Reply:
btw No need to do this in this PR we can improve after.